src/mailadm/bot.py

```diff
@@ -0,0 +1,121 @@
+import sys
+import sqlite3
+import time
+
+import deltachat
+from deltachat import account_hookimpl
+from mailadm.db import DB, get_db_path
+from mailadm.commands import add_user, add_token, list_tokens
+import os
+from threading import Event
+
+
+class SetupPlugin:
+    def __init__(self, admingrpid):
+        self.member_added = Event()
+        self.admingrpid = admingrpid
+        self.message_sent = Event()
+
+    @account_hookimpl
+    def ac_member_added(self, chat: deltachat.Chat, contact, actor, message):
+        assert chat.num_contacts() == 2
```

Comment:
not robust enough i think: if for some reason a second person tries to join the admin group, then the event-thread will error and probably die, and become dysfunctional.  If "chat.num_contacts() > 2" then a message should be send to the admin chat that there was a failed attempt at joining another address (and telling about the precise e-mail address which tried to join).

Reply:
Hm, this hook is only used during `setup-bot` so we can wait until the user actually joined the group. So we actually want to check that exactly one person joined so far, because that means the script worked. But we *want* more users to join later, so it's not really something we would warn about.

It doesn't raise an exception anymore though.